src/SubtitleOctopus.cpp

```diff
@@ -305,23 +319,27 @@ class SubtitleOctopus {
         ass_set_margins(ass_renderer, top, bottom, left, right);
     }
 
-    int getEventCount() {
+    int getEventCount() const {
         return track->n_events;
     }
 
     int allocEvent() {
+        free(m_is_event_animated);
+        m_is_event_animated = NULL;
```

Comment:
Why not use `std::vector` instead of low-level memory calls?

Reply:
Because reasons :) I mean, `std::vector` most likely would overprovision a little, taking more memory than needed, and in most scenarios this is a one-time allocation (i.e. once per one video being played).

Vector provides a few nice things (like auto-increasing upon pushing), neither of which I need here. The only one is semi-automatic memory management, and this code is easy enough to do without it.

Reply:
Oh, and I'm not sure about other libraries, but `libass` is pure C, so it doesn't bring `libstdc++` with itself, and I think that other libs may, too. So I don't want to bring the whole C++ runtime just to save like 5 lines of manual management...